stm32cube/stm32h7xx/drivers/include/stm32h7xx_ll_adc.h

```diff
@@ -6259,19 +6263,44 @@ __STATIC_INLINE uint32_t LL_ADC_GetOverSamplingDiscont(ADC_TypeDef *ADCx)
   */
 __STATIC_INLINE void LL_ADC_ConfigOverSamplingRatioShift(ADC_TypeDef *ADCx, uint32_t Ratio, uint32_t Shift)
 {
-  MODIFY_REG(ADCx->CFGR2, (ADC_CFGR2_OVSS | ADC_CFGR2_OVSR), (Shift | (((Ratio - 1UL) << ADC_CFGR2_OVSR_Pos))));
+#if defined(ADC_VER_V5_V90)
```

Comment:
I don't think STMicroelectronics developers will change the API (they will keep backward compatibility).
This mean they will keep ratio as range 1..1024  at least for H7 without ADC_VER_V5_V90 (and for ADC_VER_V5_V90 on ADC1/2)
So I would prefer to keep original code in the 'else'.
And in the 'if' (ADC_VER_V5_V90 on ADC3) I propose to do like other series (like G0, L4): use LL defines like LL_ADC_OVS_RATIO_128 as input parameter, thus no shift in LL .
Of course this imply to adapt stm32 zephyr driver accordingly.

Reply:
OK, so we decide _NOT_  to change the `LL_ADC_ConfigOverSamplingRatioShift() `for stm32H7
--> keeping the 'ratio' as a value from 1..1024 for ADC1/2 and from 1..256 for ADC3.
It means that the zephyr driver for stm32H7 will give the 2^oversampling (and no more the oversampling value)
--> changing the zephyr adc_stm32.c driver accordingly.

For the stm32H7, the ADC 3 of ADC_VER_V5_V90 : the OVSR[2:0] bitfield is 'Ratio' (1UL << sequence->oversampling)
Ratio = 256 (sequence->oversampling =8), oversampling by 256 ; OVSR[2:0] = 0b111
Ratio = 64 (sequence->oversampling = 6), oversampling by 64 ; OVSR[2:0] = 0b101
Ratio = 8 (sequence->oversampling = 3), oversampling by 8 ; OVSR[2:0] = 0b010

for other ADC: the OSVR[9:0] bitfield is ((1UL << Ratio) - 1UL) << ADC_CFGR2_OVSR_Pos
Ratio = 1024 (sequence->oversampling =10), oversampling by 1024 ; OSVR[9:0] = 1023
Ratio = 256 (sequence->oversampling =8), oversampling by 256 ; OSVR[9:0] = 255
Ratio = 64 (sequence->oversampling =6), oversampling by 64 ; OSVR[9:0] = 63
Ratio = 8 (sequence->oversampling =3) oversampling by 8 ; OSVR[9:0] = 7